pkg/apis/vsphere/v1/template_types.go

```diff
@@ -0,0 +1,13 @@
+package v1
+
+// anxcloud:object:hooks=PaginationSupportHook
+
+// Template represents a vSphere template used for vm provisioning
+type Template struct {
+	Identifier string `json:"id" anxcloud:"identifier"`
+	Name       string `json:"name"`
+	Bit        string `json:"bit"`
+	Build      string `json:"build"`
+	Location   string
+	Type       string
```

Comment:
The constant `Template` would clash with the `Template` API struct. The engine supports "templates" and "from_scratch" as valid inputs. Maybe `Templates` instead of `Template`?

Reply:
oh right ^^'

what about `TypeTemplate` and `TypeFromScratch`?